pkg/kubelet/server/portforward/websocket.go

```diff
@@ -95,49 +107,77 @@ func handleWebSocketStreams(req *http.Request, w http.ResponseWriter, portForwar
 		return err
 	}
 	defer conn.Close()
-
-	streams[dataChannel].Write([]byte{})
-	streams[errorChannel].Write([]byte{})
-
+	streamPairs := make([]*websocketStreamPair, len(opts.ports))
+	for i, _ := range streamPairs {
+		streamPair := websocketStreamPair{
+			port:        opts.ports[i],
+			dataStream:  streams[i*2+dataChannel],
+			errorStream: streams[i*2+errorChannel],
+		}
+		streamPairs[i] = &streamPair
+
+		streamPair.dataStream.Write([]byte{})
+		streamPair.errorStream.Write([]byte{})
+	}
 	h := &websocketStreamHandler{
-		conn:      conn,
-		port:      opts.port,
-		streams:   streams,
-		pod:       podName,
-		uid:       uid,
-		forwarder: portForwarder,
+		conn:        conn,
+		streamPairs: streamPairs,
+		pod:         podName,
+		uid:         uid,
+		forwarder:   portForwarder,
 	}
-
 	h.run()
 
 	return nil
 }
 
+// websocketStreamPair represents the error and data streams for a port
+// forwarding request.
+type websocketStreamPair struct {
+	port        uint16
+	dataStream  io.ReadWriteCloser
+	errorStream io.WriteCloser
+}
+
 // websocketStreamHandler is capable of processing a single port forward
 // request over a websocket connection
 type websocketStreamHandler struct {
-	conn      *wsstream.Conn
-	port      uint16
-	streams   []io.ReadWriteCloser
-	pod       string
-	uid       types.UID
-	forwarder PortForwarder
+	conn        *wsstream.Conn
+	ports       []uint16
+	streamPairs []*websocketStreamPair
+	pod         string
+	uid         types.UID
+	forwarder   PortForwarder
 }
 
 // run invokes the websocketStreamHandler's forwarder.PortForward
 // function for the given stream pair.
 func (h *websocketStreamHandler) run() {
-	defer h.streams[dataChannel].Close()
-	defer h.streams[errorChannel].Close()
-	defer h.conn.Close()
+	wg := sync.WaitGroup{}
```

Comment:
Yes. The streams are closed in portForward and the connection is closed above in handleWebSocketStreams.